openwisp_monitoring/monitoring/base/models.py

```diff
@@ -154,48 +156,97 @@ def _get_time(self, time):
             return parse_date(time)
         return time
 
-    def check_threshold(self, value, time=None, retention_policy=None, send_alert=True):
+    def _set_is_healthy(self, alert_settings, value):
         """
-        checks if the threshold is crossed
-        and notifies users accordingly
+        Sets the value of "is_healthy" field if "value"
+        crosses threshold defined in "alert_settings".
+        Returns "True" if "is_healthy" field is changed.
+        Otherwise, returns "None".
         """
-        try:
-            alert_settings = self.alertsettings
-        except ObjectDoesNotExist:
+        crossed = alert_settings._value_crossed(value)
+        if (not crossed and self.is_healthy) or (crossed and self.is_healthy is False):
             return
+        # problem: not within threshold limit
+        elif crossed and self.is_healthy in [True, None]:
+            self.is_healthy = False
+        # ok: returned within threshold limit
+        elif not crossed and self.is_healthy in [False, None]:
+            self.is_healthy = True
+        return True
+
+    def _set_is_tolerance_healthy(
+        self, alert_settings, value, time, retention_policy, send_alert
+    ):
+        """
+        Sets the value of "is_tolerance_healthy" if "value"
+        crosses threshold for tolerance defined in alert_settings.
```

Comment:
I would try to make this dosctring a bit more readable:

Sets the value of "is_tolerance_healthy" if "value"
crosses the threshold for more than the amount of seconds defined in the alert_settings tolerance field.

Reply:
We could also write that this method is very similar to `_set_is_healthy` but it takes into account the alert settings tolerance so it's slightly different and more complex.